streams/src/test/java/org/apache/kafka/streams/state/internals/QueryableStoreProviderTest.java

```diff
@@ -88,5 +91,23 @@ public void shouldFindGlobalStores() {
         assertNotNull(storeProvider.getStore(StoreQueryParameters.fromNameAndType("global", QueryableStoreTypes.keyValueStore())));
     }
 
+    @Test
+    public void shouldReturnKVStoreWithPartitionWhenItExists() {
+        assertNotNull(storeProvider.getStore(StoreQueryParameters.fromNameAndType(keyValueStore, QueryableStoreTypes.keyValueStore()).withPartition(numStateStorePartitions - 1)));
+    }
+
+    @Test(expected = InvalidStateStoreException.class)
+    public void shouldThrowExceptionWhenKVStoreWithPartitionDoesntExists() {
+        storeProvider.getStore(StoreQueryParameters.fromNameAndType(keyValueStore, QueryableStoreTypes.keyValueStore()).withPartition(numStateStorePartitions + 1));
```

Comment:
Can we split this as follows:
```
final StoreQueryParameters parameters = (StoreQueryParameters.fromNameAndType(keyValueStore, QueryableStoreTypes.keyValueStore()).withPartition(numStateStorePartitions + 1);

final InvalidStateStoreException exception = asserThrows(
  InvalidStateStoreException.class,
  () -> storeProvider.getStore(parameters)
);
assertThat(exception.message(), equalTo("..."));
```

And remove the `(excpected = ...)` annotation.

(1) We should always limit the code that might throw the exception (eg, if `withPartition` would throw an `InvalidStateStoreException` the test should fail, but would pass in it's current setup) (2) We should always verify the exception cause -- `getStore()` could throw an `InvalidStateStoreException` or multiple reasons and we should make sure it's throwing for the reason under test.

Same below for the windowed case